tests/runner/src/omvf.rs

```diff
@@ -0,0 +1,261 @@
+    }
+
+    fn extension(&self) -> &'static str {
+        match self {
+            Self::Code | Self::Vars => "fd",
+            Self::Shell => "efi",
+        }
+    }
+
+    /// Get a user-provided path for the given OVMF file type.
+    ///
+    /// This uses an environment variable. If not present, returns `None`.
+    pub fn get_user_provided_path(self) -> Option<PathBuf> {
+        let var_name = match self {
+            Self::Code => ENV_VAR_OVMF_CODE,
+            Self::Vars => ENV_VAR_OVMF_VARS,
+            Self::Shell => ENV_VAR_OVMF_SHELL,
+        };
+        env::var_os(var_name).map(PathBuf::from)
+    }
+}
+
+pub struct OvmfPaths {
+    code: PathBuf,
+    vars: PathBuf,
+    shell: PathBuf,
+    temp_dir: Option<TempDir>,
+    temp_vars: Option<PathBuf>,
+}
+
+impl OvmfPaths {
+    pub fn code(&self) -> &PathBuf {
+        &self.code
+    }
+
+    pub fn vars(&self) -> &PathBuf {
+        self.temp_vars.as_ref().unwrap_or(&self.vars)
+    }
+
+    pub fn shell(&self) -> &PathBuf {
+        &self.shell
+    }
+
+    /// Search for an OVMF file (either code or vars).
+    ///
+    /// There are multiple locations where a file is searched at in the following
+    /// priority:
+    /// 1. Command-line arg
+    /// 2. Environment variable
+    /// 3. Prebuilt file (automatically downloaded)
+    pub fn find_ovmf_file(file_type: OvmfFileType) -> Result<PathBuf> {
+        if let Some(path) = file_type.get_user_provided_path() {
+            // The user provided an exact path to use; verify that it
+            // exists.
+            if path.exists() {
+                Ok(path)
+            } else {
+                bail!(
+                    "ovmf {} file does not exist: {}",
+                    file_type.as_str(),
+                    path.display()
+                );
+            }
+        } else {
+            let prebuilt_dir = update_prebuilt()?;
+
+            Ok(prebuilt_dir.join(format!(
+                "x86_64/{}.{}",
+                file_type.as_str(),
+                file_type.extension()
+            )))
+        }
+    }
+
+    /// Find path to OVMF files by the strategy documented for
+    /// [`Self::find_ovmf_file`].
+    pub fn find() -> Result<Self> {
+        let code = Self::find_ovmf_file(OvmfFileType::Code)?;
+        let vars = Self::find_ovmf_file(OvmfFileType::Vars)?;
+        let shell = Self::find_ovmf_file(OvmfFileType::Shell)?;
+
+        Ok(Self {
+            code,
+            vars,
+            shell,
+            temp_dir: None,
+            temp_vars: None,
+        })
+    }
+
+    /// Creates a copy with a writable, temp copy
+    pub fn with_temp_vars(&mut self) -> Result<()> {
+        let temp_dir = TempDir::new()?;
+        let temp_path = temp_dir.path();
+
+        // Make a copy of the OVMF vars file so that it can be used
+        // read+write without modifying the original. Under AArch64, some
+        // versions of OVMF won't boot if the vars file isn't writeable.
```

Comment:
Maybe reading the file and writing it to a new file would work to avoid an explicit permission change?

Reply:
I'm not sure what you mean. I am creating a copy of the original file. Or are you referring to this?
```
// Necessary, as for example on NixOS, the files are read-only inside
// the Nix store.
#[cfg(target_os = "linux")]
fs_err::set_permissions(&ovmf_vars, Permissions::from_mode(0o666))?;
```
I copied that from the uefi crates tests. I did not really look into this too much. I assume they had some issues that meant they needed to add this.

Reply:
If you use fs::copy you make a copy of the file including pernissions and xattrs. It seems that here a copy of just the file contents ignoring permissions would be a better option.

Reply:
I'm not convinced that would be any cleaner though. A simple call to `set_permissions` is cleaner in my opinion. 
Also this way we get some possible advantages from COW file systems, etc.

Reply:
At the very least make it `#[cfg(unix)]` then.